harness/determined/exec/tensorboard.py

```diff
@@ -17,10 +19,14 @@
 
 TENSORBOARD_TRIGGER_READY_MSG = "TensorBoard contains metrics"
 TRIGGER_WAITING_MSG = "TensorBoard waits on metrics"
-TICK_INTERVAL = 1
-MAX_WAIT_TIME = 600
-TB_RESPONSE_WAIT_TIME = 300
 
+TICK_INTERVAL = 1  # How many seconds to wait on each iteration of our check loop
+MAX_WAIT_TIME = 600  # How many seconds to wait for the first metric file to download
+TB_RESPONSE_WAIT_TIME = 300  # How many seconds to wait for TensorBoard to initially start up
+WORK_QUEUE_MAX_SIZE = 20  # Size of the threading work queue for fetching
+FULL_ITERATION_SLEEP_TIME = 20  # How long to wait between a full iteration run (in seconds)
+NUM_FETCH_THREADS = 5  # Number of fetching threads to run concurrently
+READY_SIGNAL_DELAY = 7  # How many seconds to wait before sending the ready signal
```

Comment:
what's the rationale behind some of the constants? like `READY_SIGNAL_DELAY` or number of threads?

Reply:
We delay sending the ready signal because if we send it immediately after syncing the first file, tensorboard itself needs to do some bookkeeping to sync the file. We wanted to make sure that we can signal ready early (rather than after every file is uploaded like in the original version of this), but not so early that we get erroneous "no data" screens.

I was toying with the idea of reading the html of the tensorboard page to determine that it's actually showing data before sending the ready signal, but I'm concerned it would be highly fragile to changes in future (or past) versions of tensorboard. I opted to make it a simple delay of 5 seconds and in practice, this seems to work. 

Worst case, it shows the "no data" screen regardless, which it will update itself away from in about 5 seconds

Reply:
Number of threads is honestly arbitrary, I just didn't want to spin up a huge amount of them to not overwhelm the machine (we use very weak nodes like t2's on AWS for this).